ethereumj-core/src/main/java/org/ethereum/net/eth/handler/Eth62.java

```diff
@@ -792,6 +792,13 @@ public synchronized void dropConnection() {
         disconnect(USELESS_PEER);
     }
 
+    @Override
+    public synchronized boolean setStatus(SyncState syncState) {
+        if (!sentHeaders.isEmpty() && !headerRequests.isEmpty()) return false;
```

Comment:
@Nashatyrev I did it that way first time, but we need to return peer to IDLE if we locked it but there is no need to use in the end, so I added such checking instead